### ISSUE TITLE
Potential bug with 'start_points' argument of 'pyplot.streamplot'

### ISSUE THREAD

**Issue by muahah** [matplotlib/matplotlib#6002]
Using `pyplot.streamplot` with `start_points` argument lead to streamline misplacement : 
![test](https://cloud.githubusercontent.com/assets/6860164/13037132/48d1f2b8-d37a-11e5-94f9-c3acd490b8ca.png)
Source code : 

```
import numpy as np
import matplotlib.pyplot as plt

# Data
x = np.linspace(-10, 10, 10)
y = np.linspace(-10, 10, 10)
X, Y = np.meshgrid(x, y)
U = X*0 + 1
V = X*0
start_points = [[0, 0]]

# Base streamline plot
plt.figure()
sp1 = plt.streamplot(x, y, U, V, color=[.5]*3)

# Streamline plot with 'start_points' argument
sp2 = plt.streamplot(x, y, U, V, start_points=start_points,
                     color='r')
plt.plot(*start_points[0], marker='o', label="Starting point")
plt.plot([], [], color='r', label="Associated streamline")

# Legend and limits
plt.xlim(-10, 10)
plt.ylim(-10, 10)
plt.legend(numpoints=1)

plt.show()
```

**Comment by tacaswell** [comment]
Looking at the source it looks like the starting point is used to seed the 'mask' layer, but I am not familiar with this code.

@Cadair There is an explicit shift to the start points you added in https://github.com/matplotlib/matplotlib/pull/4714  Can you take a look at or comment on this?

attn @tonysyu

**Comment by Cadair** [comment]
Sorry I have not had a chance to look at this. Was this affected by #6011 ?

@tacaswell I do not know what the explicit shift in the start points does, I am assuming past @cadair put it there for a reason, but he is requently proven to do silly things!

I will try and take a close look, at the code but I can't promise when.

**Comment by muahah** [comment]
Indeed, this issue should have been fixed by #6011, I just forgot to close it.
But an informed opinion on what have been done in #6011 concerning `starting_points` (commit  71b9ae8) would be great.
